src/Component.js

```diff
@@ -0,0 +1,114 @@
+import Observable from "./utils/Observable";
+import style from "./CleverDiagram.css";
+import * as d3 from "d3";
+
+class Component {
+
+	constructor(
+		className = "component"
+	) {
+		this._container = null;
+		this._className = className;
+
+		this._observable = new Observable([
+			"enter",
+			"leave",
+			"click"
+		]);
+
+		this._lastData = null;
+		this._svg = null;
+	}
+
+	on(eventName, handler) {
+		this._observable.on(eventName, handler);
+		return this;
+	}
+
+	off(eventName, handler) {
+		this._observable.off(eventName, handler);
+		return this;
+	}
+
+	get className() {
+		return this._className;
+	}
+
+	get container() {
+		return this._container;
+	}
+
+    get observable() {
```

Comment:
same indentation here?

Reply:
Yes, we're mixing tabs and spaces. Please select one option and use it everywhere. ESLint should throw an error if configured...